src/engine/ui.cpp

```diff
@@ -95,6 +95,44 @@ struct gui : guient
     static int curdepth, curlist, xsize, ysize, curx, cury, fontdepth, mergelist, mergedepth;
     static bool hitfx, skinfx, cursorfx;
 
+    // Example: {3, 2} means that the current object (button/slider/whatever) is the 2nd child of the 3rd top-level child.
+    // The exact numbers don't matter as long as they are unique for each object.
+    static vector<int> curchildpath;
```

Comment:
why not use `std::vector`?

Reply:
Because vector is used in other places. It could be confusing to mix the two.

Reply:
We wanna switch to using STL datatypes and functions instead of the custom implementation, IIRC the custom vector has severe memory problems, but also it's way easier to use the STL for everybody because everybody usually knows how it behaves, if you look at PRs that have been merged so far, they all use `std::vector` instead of `vector` unless it has a serious advantage to use `vector`